internal/jobs/pubkey_upload_aws_job.go

```diff
@@ -91,7 +91,11 @@ func handlePubkeyUploadAWS(ctx context.Context, args *PubkeyUploadAWSTaskArgs) e
 	pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
```

Comment:
:warning: Handle would be a name of the key now. Can we handle it? We will need to do something about our DB values on stage (cleanup of the Pubkey resources comes to mind as simple solution).

Reasoning: we are passing a name to the `RunInstances`, so it should be the handle, what for we keep the Handle if not to be used to be passed to `RunInstances`.
@lzap @adiabramovitch

Reply:
When I designed this, I was more focused on the job scheduler part than on the AWS part, so I sort of mindlessly created this "upload pubkey" job. I think it is the time to revisit this from the ground up and rethink this. Few points:

* The "aws handle" field does not have to be the "identifier" we use for keys, it can serve as an informative field too. I very much like the idea of searching pubkeys by fingerprint because it looks like on AWS this is what the API supports and it also enforces uniqueness therefore it is guaranteed to find 0 or 1 entry.
* The job is poorly named, I think we should think about our job scheduler as a small "ansible", it should be idempotent. Therefore I think we should rename it to EnsurePubkeyAWS or something like that - it should do whatever must be done to have that pubkey on the AWS account.
* It is absolutely no problem to perform more AWS API requests in a background job, we have all the time in the world, we can slow down jobs if we hit any API limits. So whoever want's to redesign this, they can even simply just query AWS first, if a key is found, then we can compare our handle and update it (or maybe it can be completely dropped we don't know if this is possible on other clouds). If it's not found, upload it and update our records.
* I think it is a good time to think about documenting each job, I suggest to have something close to the code, so code comment describing what the job should do would be a really nice idea. We can maybe start writing the new "job story" on the next tech meeting and once we agree we have a good approach that encapsulates all the edge cases, we can go ahead and implement it.